craft_parts/plugins/go_plugin.py

```diff
@@ -121,20 +122,26 @@ def get_build_commands(self) -> list[str]:
         options = cast(GoPluginProperties, self._options)
 
         # Matches go-use plugin expectation.
-        workspace_dir = self._part_info.project_info.dirs.parts_dir
-        workspace = workspace_dir / "go.work"
-
-        if workspace.exists():
-            init_cmd = f"go work use {self._part_info.part_build_dir}"
+        dependencies_dir: pathlib.Path = self._part_info.backstage_dir / "go-use"
+        if dependencies_dir.is_dir():
+            setup_cmds = [
+                "go work init .",
+                "go work use .",
+                *(
+                    f"go work use {dependencies_dir}/{dep_part}"
+                    for dep_part in self._part_info.part_dependencies
```

Comment:
I understand the logic behind using the part's dependencies here, but
a) I am not sure that this is the full set of dependencies; I suspect it might just be the names in [after](https://github.com/canonical/craft-parts/blob/main/craft_parts/parts.py#L418)
b) This is a change from the previous behavior that we should discuss - very likely it will break existing projects that don't declare "correct" part ordering